numba/core/byteflow.py

```diff
@@ -651,24 +789,46 @@ def op_YIELD_VALUE(self, state, inst):
         state.append(inst, value=val, res=res)
         state.push(res)
 
-    def op_RAISE_VARARGS(self, state, inst):
-        in_exc_block = any([
-            state.get_top_block("EXCEPT") is not None,
-            state.get_top_block("FINALLY") is not None
-        ])
-        if inst.arg == 0:
-            exc = None
-            if in_exc_block:
+    if PYVERSION == (3, 11):
+        def op_RAISE_VARARGS(self, state, inst):
+            if inst.arg == 0:
+                exc = None
                 raise UnsupportedError(
                     "The re-raising of an exception is not yet supported.",
                     loc=self.get_debug_loc(inst.lineno),
                 )
-        elif inst.arg == 1:
-            exc = state.pop()
-        else:
-            raise ValueError("Multiple argument raise is not supported.")
-        state.append(inst, exc=exc)
-        state.terminate()
+            elif inst.arg == 1:
+                exc = state.pop()
+            else:
+                raise ValueError("Multiple argument raise is not supported.")
+            state.append(inst, exc=exc)
+
+            if state.has_active_try():
+                self._adjust_except_stack(state)
+            else:
+                state.terminate()
+
+    elif PYVERSION < (3, 11):
+        def op_RAISE_VARARGS(self, state, inst):
+            in_exc_block = any([
+                state.get_top_block("EXCEPT") is not None,
+                state.get_top_block("FINALLY") is not None
+            ])
+            if inst.arg == 0:
+                exc = None
+                if in_exc_block:
+                    raise UnsupportedError(
+                        "The re-raising of an exception is not yet supported.",
+                        loc=self.get_debug_loc(inst.lineno),
+                    )
+            elif inst.arg == 1:
+                exc = state.pop()
+            else:
+                raise ValueError("Multiple argument raise is not supported.")
+            state.append(inst, exc=exc)
+            state.terminate()
+    else:
+        raise NotImplementedError
```

Comment:
Several other places that used the if = 3.11, elif < 3.11 else NotImplemented pattern in this file used NotImplementedError(PYVERSION), so should this line also have the (PYVERSION) argument?